src/main/java/fi/otavanopisto/kuntaapi/server/integrations/management/ManagementNewsArticleEntityUpdater.java

```diff
@@ -0,0 +1,175 @@
+  
+  @Inject
+  private ModificationHashCache modificationHashCache;
+  
+  @Resource
+  private TimerService timerService;
+
+  private boolean stopped;
+  private List<NewsArticleIdUpdateRequest> queue;
+
+  @PostConstruct
+  public void init() {
+    queue = Collections.synchronizedList(new ArrayList<>());
+  }
+
+  @Override
+  public String getName() {
+    return "management-news";
+  }
+
+  @Override
+  public void startTimer() {
+    startTimer(TIMER_INTERVAL);
+  }
+
+  private void startTimer(int duration) {
+    stopped = false;
+    TimerConfig timerConfig = new TimerConfig();
+    timerConfig.setPersistent(false);
+    timerService.createSingleActionTimer(duration, timerConfig);
+  }
+
+  @Override
+  public void stopTimer() {
+    stopped = true;
+  }
+  
+  @Asynchronous
+  public void onOrganizationIdUpdateRequest(@Observes NewsArticleIdUpdateRequest event) {
+    if (!stopped) {
+      NewsArticleId newsArticleId = event.getId();
+      
+      if (!StringUtils.equals(newsArticleId.getSource(), ManagementConsts.IDENTIFIER_NAME)) {
+        return;
+      }
+      
+      if (event.isPriority()) {
+        queue.remove(event);
+        queue.add(0, event);
+      } else {
+        if (!queue.contains(event)) {
+          queue.add(event);
+        }
+      }
+    }
+  }
+  
+  @Timeout
+  public void timeout(Timer timer) {
+    if (!stopped) {
+      if (!queue.isEmpty()) {
+        NewsArticleIdUpdateRequest updateRequest = queue.remove(0);
+        updateManagementPost(updateRequest.getOrganizationId(), updateRequest.getId());
+      }
+
+      startTimer(SystemUtils.inTestMode() ? 1000 : TIMER_INTERVAL);
+    }
+  }
+  
+  private void updateManagementPost(OrganizationId organizationId, NewsArticleId newsArticleId) {
+    DefaultApi api = managementApi.getApi(organizationId);
+    
+    ApiResponse<Post> response = api.wpV2PostsIdGet(newsArticleId.getId(), null);
+    if (response.isOk()) {
+      updateManagementPost(api, response.getResponse());
+    } else {
+      logger.warning(String.format("Find organization %s post %s failed on [%d] %s", organizationId.getId(), newsArticleId.toString(), response.getStatus(), response.getMessage()));
+    }
+  }
+  
+  private void updateManagementPost(DefaultApi api, Post managementPost) {
+    NewsArticleId newsArticleId = new NewsArticleId(ManagementConsts.IDENTIFIER_NAME, String.valueOf(managementPost.getId()));
+
+    Identifier identifier = identifierController.findIdentifierById(newsArticleId);
+    if (identifier == null) {
+      identifier = identifierController.createIdentifier(newsArticleId);
+    }
+    
+    modificationHashCache.put(identifier.getKuntaApiId(), createPojoHash(managementPost));
+    
+    if (managementPost.getFeaturedMedia() != null && managementPost.getFeaturedMedia() > 0) {
+      updateFeaturedMedia(api, managementPost.getFeaturedMedia()); 
+    }
+  }
+  
+  private void updateFeaturedMedia(DefaultApi api, Integer featuredMedia) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "updateFeaturedMedia" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)